### PR TITLE
feature(pictures): #CO-1817 check bad request when create ticket

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

Currently, if you created a ticket and it returned a bad request, you could no longer create a ticket.
Now, it returns an error, but you can still recreate a ticket.

## Checklist tests

- Custom route POST `/support/ticket` to send a response 400 and try to create two tickets

## Issue ticket number and link

[https://jira.support-ent.fr/browse/CO-1817](https://jira.support-ent.fr/browse/CO-1817)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [X] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/controllers/controller.ts [hunk 1]
```diff
@@ -445,6 +445,16 @@ export const SupportController: Controller = ng.controller('SupportController',
...
 				$scope.ticket.processing = false;
+
+				const e400Callback = function (result) {
+					if (result && result.error) {
+						notify.error(result.error);
+					} else {
+						notify.error('support.error.create.ticket');
+					}
+					$scope.newTicket();
+				};
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/controllers/controller.ts:456]: 🤔 y'aurait pas moyen de rendre la méthode un peut plus générique, afin qu'on puisse réutiliser cette partie ailleurs ?

File: src/main/resources/public/ts/models/model.ts [hunk 2]
```diff
@@ -82,14 +82,22 @@ model.isBugTrackerCommDirect = function(callback){
...
-		if(typeof callback === 'function'){
-			callback();
-		}
-	}.bind(this));
+models.Ticket.prototype.createTicket = function(data, callback, badRequestCallback) {
+	http().postJson('/support/ticket', data)
+		.done(function(result){
+				this.updateData(result);
+				model.tickets.push(this);
+				if(typeof callback === 'function'){
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/models/model.ts:90]: indentation !

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/models/model.ts (hunk 2 above)
Comment (Mattox0) [reply src/main/resources/public/ts/models/model.ts:90]: j'avais suivi la même indentation que la fonction `escalateTicket` qui se situe en dessous, je change ça

File: src/main/resources/public/ts/models/model.ts [hunk 3]
```diff
@@ -82,14 +82,20 @@ model.isBugTrackerCommDirect = function(callback){
...
-		if(typeof callback === 'function'){
-			callback();
-		}
-	}.bind(this));
+models.Ticket.prototype.createTicket = function(data, callback, badRequestCallback) {
+	http().postJson('/support/ticket', data)
+		.done(function(result){
+			this.updateData(result);
+			model.tickets.push(this);
+			if(typeof callback === 'function'){
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/models/model.ts:90]: indentation !

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

File: src/main/resources/public/ts/models/model.ts [hunk 4]
```diff
@@ -82,14 +82,20 @@ model.isBugTrackerCommDirect = function(callback){
...
-	http().postJson('/support/ticket', data).done(function(result){
-		this.updateData(result);
-		model.tickets.push(this);
-		if(typeof callback === 'function'){
-			callback();
-		}
-	}.bind(this));
+models.Ticket.prototype.createTicket = function(data, callback, badRequestCallback) {
+	http().postJson('/support/ticket', data)
+		.done(function(result){
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/models/model.ts:87]: indentation !

File: src/main/resources/public/ts/models/model.ts [hunk 5]
```diff
@@ -82,14 +82,20 @@ model.isBugTrackerCommDirect = function(callback){
...
+			this.updateData(result);
+			model.tickets.push(this);
+			if(typeof callback === 'function'){
+				callback();
+			}
+		}.bind(this))
+		.e400(function(){
+			if(typeof badRequestCallback === 'function'){
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/models/model.ts:95]: indentation !

File: src/main/resources/public/ts/models/model.ts [hunk 6]
```diff
@@ -82,14 +82,20 @@ model.isBugTrackerCommDirect = function(callback){
...
+models.Ticket.prototype.createTicket = function(data, callback, badRequestCallback) {
+	http().postJson('/support/ticket', data)
+		.done(function(result){
+			this.updateData(result);
+			model.tickets.push(this);
+			if(typeof callback === 'function'){
+				callback();
+			}
+		}.bind(this))
+		.e400(function(){
```
Comment (qpic-cgi) [inline src/main/resources/public/ts/models/model.ts:94]: indentation !

**Review by qpic-cgi — CHANGES_REQUESTED** [review]

**Review by qpic-cgi — APPROVED** [review]

**Review by qperie — APPROVED** [review]

**Pull request merged**